packages/nimble-components/src/rich-text/editor/index.ts

```diff
@@ -354,6 +406,40 @@ export class RichTextEditor extends RichText implements ErrorPattern {
         return Array.from(mentionedHrefs);
     }
 
+    public getMentionExtensionConfig(): MentionExtensionConfiguration[] {
+        return this.configuration instanceof EditorConfiguration
+            ? this.configuration.mentionExtensionConfig
+            : [];
+    }
+
+    public activateMention(event: CustomEvent<MentionDetail>): void {
+        if (this.activeMentionCommand) {
+            this.activeMentionCommand({
+                href: event.detail.href,
+                label: event.detail.displayName
+            });
+        }
+    }
+
+    /**
+     * @internal
+     */
+    public override handleChange(source: unknown, args: unknown): void {
+        if (
+            source instanceof MentionInternals
+            && MentionExtensionConfiguration.isObservedMentionInternalsProperty(
+                args
+            )
+        ) {
+            const mentionExtensionConfig = this.getMentionExtensionConfigFromCharacter(source.character);
+            if (mentionExtensionConfig) {
+                mentionExtensionConfig.label = source.buttonLabel ?? '';
+            }
+            return;
+        }
+        super.handleChange(source, args);
```

Comment:
I dunno if this is trying to be an optimization or something but don't be clever. Just recreate the config. This will write to the configuration property which is observable and shouuld refresh other bindings.

```suggestion
        if (
            source instanceof MentionInternals
            && MentionExtensionConfiguration.isObservedMentionInternalsProperty(
                args
            )
        ) {
            this.configuration = this.createConfig();
        } else {
            super.handleChange(source, args);
        }
        
```

Reply:
Updated